libfortres/include/fortres/stringref.fmod

```diff
@@ -82,8 +82,8 @@
 
 !_PROC_EXPORT(stringref_of_buf)
         function stringref_of_buf( fstrBuf ) result(strRef)
-          character(len=*),  intent(out) :: fstrBuf
-          type (StringRef_t)             :: strRef
+          character(len=*),  intent(in) :: fstrBuf
+          type (StringRef_t)            :: strRef
           strRef = stringref_of_str_trim( fstrBuf, .false. )
         end function

```

Comment:
We don't want to change the contents of the given buffer here.
Should we really do something like a self assignment or something to avoid the warning?  :-?